File: Java/Medium/Accepted/minimum-number-of-swaps-to-make-the-string-balanced_2.java
class Solution {
    public int minSwaps(String s) {
        
        int ans = 0;
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '[') {
                count++;
            } else {
                count--;
            }
            if (count < 0) {
                ans++;
                count = 1;
            }
        }
        return ans;
    }
}

File: Rust/Medium/WrongAnswer/minimum-number-of-swaps-to-make-the-string-balanced_2.rs
impl Solution {
    pub fn min_swaps(s: String) -> i32 {
        
        let mut res = 0;
        let mut count = 0;
        for c in s.chars() {
            if c == '[' {
                count += 1;
            } else {
                if count == 0 {
                    res += 1;
                } else {
                    count -= 1;
                }
            }
        }
        res
    }
}